modules/nextflow/src/main/groovy/nextflow/container/PodmanBuilder.groovy

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2013-2019, Centre for Genomic Regulation (CRG)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.container
+import groovy.transform.CompileStatic
+/**
+ * Helper methods to handle podman containers
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ * @author tbugfinder <github@online.ms>
+ */
+@CompileStatic
+class PodmanBuilder extends ContainerBuilder<PodmanBuilder> {
+
+    private boolean sudo
```

Comment:
The use of `sudo` as always been a bad practice with docker and with podman should not be needed in any case. Could this be removed ?

Reply:
One limit of rootless Podman is that cpuset cannot be applied. Therefore the possibility to still use Podman with sudo is included.
On the other hand Podman integration could be marked as <beta> until all features are supported.

Reply:
I see, but it must be noted that cpuset is really a corner case and I don't think there's a concrete use in nextflow for it. Moreover, podman is useful because allows the user to run container in the user space without the need to escalate root. From this point of view makes no sense to give the `sudo` option. 

Maybe it could be useful instead to support `--cpus`. Does it require sudo as well?

Reply:
`--cpus` can be used in rootless mode.
I agree on removing sudo.

Reply:
It could be part of engineOptions anyway.

Reply:
commit 3fb4e751c299b914e5f8f6c5d89e5e79a3fa63b2
Should the name be different than `cpus`?

Reply:
let's ignore it for now, NF is not to handle at this point

Reply:
Removed sudo, cpus and adopted the Tests.

Reply:
excellent